pkg/kp/consulutil/watch.go

```diff
@@ -208,3 +209,95 @@ func WatchNewKeys(pairsChan <-chan api.KVPairs, onNewKey NewKeyHandler, done <-c
 		}
 	}
 }
+
+type WatchedChanges struct {
+	Created api.KVPairs
+	Updated api.KVPairs
+	Deleted api.KVPairs
+}
+
+// WatchDiff watches a Consul prefix for changes and categorizes them
+// into create, update, and delete
+func WatchDiff(
+	prefix string,
+	clientKV ConsulLister,
+	outChanges chan<- *WatchedChanges,
```

Comment:
I know you're imitating `WatchPrefix`, but I'd actually prefer that this function initializes `outChanges` and returns it, doing the work inside of a goroutine. That's more conventional in Go, and we were just bad at following conventions back in the day :)